### PR TITLE
Update README.md to make it clear people should avoid direct API key in code

### PR DESCRIPTION
**Pull request by logankilpatrick**

### CODE REVIEW

**Comment by ADefWebserver** [comment]
Please don't add this. If a user ads their settings to the .secrets file then it is totally safe to us ethe constructor. I have a book coming out soon that uses code like this: https://github.com/SyncfusionSuccinctlyE-Books/OpenAI-Succinctly/blob/main/Pages/Index.razor where I describe how to safely do this. Thanks!

**Comment by StephenHodgson** [comment]
I, for one, use it on my back end a lot. This library is meant to be used in mostly back in scenarios, so I don't feel like explicitly not recommending it is explicitly required.

There is a proxy that can transform your authenticated OAuth token for the key. That's been documented.

File: README.md [hunk 1]
```diff
@@ -140,6 +128,18 @@ Use your system's environment variables specify an api key and organization to u
 var api = new OpenAIClient(OpenAIAuthentication.LoadFromEnv());
 ```
 
+#### Pass keys directly with constructor (no recommended)
```
Comment (StephenHodgson) [inline README.md:131]: Changing the name of this section will break the navigation in the table of contents.

Comment (StephenHodgson) [reply README.md:131]: I'm fine with adding additional text below the section divider to say that this is meant to be used to get started quickly and to test, but should not be recommended for production.

**Comment by StephenHodgson** [comment]
@logankilpatrick do you have the ability to add this library and my unity one to the [list of OpenAI community libraries](https://platform.openai.com/docs/libraries/community-libraries)?

File: README.md (hunk 1 above)
Comment (logankilpatrick) [reply README.md]: Updated the name so it won't break navigation

**Comment by logankilpatrick** [comment]
> do you have the ability to add this library and [my unity one](https://github.com/RageAgainstThePixel/com.openai.unity) to the [list of OpenAI community libraries](https://platform.openai.com/docs/libraries/community-libraries)?

Yes, trying to see if this passes the bar, I don't think hard coding API keys is the best practice here, would prefer to suggest other options first.

**Comment by ADefWebserver** [comment]
You can add the Organization ID and API key (obtained earlier) in this file, however, your keys could accidently get compromised if you checked the file into source control. The recommended method of storing your keys is to use the secrets file.

image

Right-click the Project node and select Manage User Secrets.
Change all the code to the following, updating ** Your OpenAI Organization ** with your Organization ID and ** Your OpenAI ApiKey ** with your API key:

```
{
  "OpenAIServiceOptions": {
    "Organization": "** Your OpenAI Organization **",
    "ApiKey": "** Your OpenAI ApiKey **"
  }
}
```
 
image


To retrieve the setting values, add the following code to the Program.cs file under the var builder = WebApplication.CreateBuilder(args); line:

```
builder.Configuration.AddJsonFile(
    "appsettings.json", optional: true, reloadOnChange: true
    )
    .AddEnvironmentVariables()
    .AddUserSecrets(Assembly.GetExecutingAssembly(), true);
```

**Comment by StephenHodgson** [comment]
So one last comment, is that the order of precedence of how we load API keys. moving the section order reflected this but does not any longer.

**Pull request merged**